util/parallel.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018-present The Alive2 Authors.
+// Distributed under the MIT license that can be found in the LICENSE file.
+
+#include "util/parallel.h"
+#include "util/compiler.h"
+#include <cassert>
+#include <fcntl.h>
+#include <fstream>
+#include <regex>
+#include <sstream>
+#include <string>
+#include <sys/wait.h>
+#include <unistd.h>
+
+// TODO: read this carefully and make sure it's safe when pids are
+// reused
```

Comment:
uhm, there's no issue ATM since you keep everything in a vector and vector idxs are used. So all good, right?

Reply:
well, I had meant to sit down for 5 minutes and make sure I believed this argument. I think I do, I removed the comment.

but also if we were compiling a large file this array could get quite large and it's awkward to reuse slots. I was thinking about replacing it with an unordered_map indexed by PID. that should be totally fine as long as the index only lives for as long as the associated process.